services/distribution/src/main/java/app/coronawarn/server/services/distribution/dgc/DigitalGreenCertificateToProtobufMapping.java

```diff
@@ -188,10 +192,16 @@ private ValueSet read(String valueSetId, ConfigValueProvider configGetter, Strin
    * @return The ValueSet or empty.
    */
   private Optional<ValueSet> getValueSet(String valueSetId) {
-    Optional<String> hash = getValueSetHash(valueSetId);
-    if (hash.isPresent()) {
-      return dccClient.getValueSet(hash.get());
+    Optional<String> hash;
+    try {
+      hash = getValueSetHash(valueSetId);
+      if (hash.isPresent()) {
+        return dccClient.getValueSet(hash.get());
+      }
+    } catch (DigitalCovidCertificateException e) {
+      return Optional.empty();
```

Comment:
Why not simply declare the thrown exception? Would make all else easier, or not?